### PR TITLE
doc: miscellaneous improvements to std::path::Path examples

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

File: src/libstd/path.rs [hunk 1]
```diff
@@ -1424,10 +1427,10 @@ impl Path {
     /// ```
     /// use std::path::Path;
     ///
-    /// let path = Path::new("hello_world.rs");
-    /// let filename = "hello_world.rs";
+    /// let path = Path::new("foo.txt");
+    /// let os_str = std::ffi::OsStr::new("foo.txt");
```
Comment (steveklabnik) [inline src/libstd/path.rs]: This is more idiomatically `use std::ffi::OsStr; OsStr::new`

**Comment by steveklabnik** [comment]
r=me after a nit

**Comment by tshepang** [comment]
\0/

**Comment by steveklabnik** [comment]
https://github.com/rust-lang/rust/pull/25734#issuecomment-104924063

**Comment by tshepang** [comment]
fixed

**Comment by steveklabnik** [comment]
Thanks :)

**Pull request merged**